src/Revision.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq;
+
+use Rinq\Exception\FrozenAttributeException;
+use Rinq\Exception\StaleFetchException;
+use Rinq\Exception\StaleUpdateException;
+use Rinq\Ident\Reference;
+
+/**
+ * Revision represents a specific revision of session.
+ *
+ * Revision is the sole interface for manipulating a session's attribute table.
+ *
+ * The underlying session may be "local", i.e. owned by a peer running in this
+ * process, or "remote", owned by a different peer.
+ *
+ * For remote sessions, operations may require network IO. Deadlines are
+ * honoured for all methods that accept a context.
```

Comment:
Switch `Deadlines` to `Timeouts` here, since our PHP `Context` knock-off doesn't have an actual deadline.